src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZFile.java

```diff
@@ -981,8 +979,13 @@ private void readFilesInfo(final ByteBuffer header, final Archive archive) throw
                     int nextName = 0;
                     for (int i = 0; i < names.length; i += 2) {
                         if (names[i] == 0 && names[i+1] == 0) {
-                            files[nextFile++].setName(new String(names, nextName, i-nextName, StandardCharsets.UTF_16LE));
+                            String fName = new String(names, nextName, i - nextName, StandardCharsets.UTF_16LE);
+                            if (files[nextFile] == null) {
+                                files[nextFile] = new SevenZArchiveEntry();
+                            }
+                            files[nextFile].setName(fName);
```

Comment:
I think we do not need a variable `fName` here :

```
if (files[nextFile] == null) {
    files[nextFile] = new SevenZArchiveEntry();
}
files[nextFile].setName(new String(names, nextName, i - nextName, StandardCharsets.UTF_16LE));
```

Reply:
Yes, I got rid of the variable in the latest commit